snapfaas/src/labeled_fs/mod.rs

```diff
@@ -39,6 +38,8 @@ lazy_static::lazy_static! {
 pub enum Error {
     BadPath,
     Unauthorized,
+    BadTargetLabel,
+    UidCollision,
```

Comment:
It's seems like a red flag that these errors are _only_ ever handled in `sffs`, and not in the syscalls themselves. Handling them should be a good indication of whether they are reasonable errors or not (presumably in the `UidCollision` case we should just retry? In which case, maybe no need for the error? Note that the previous implementation avoids this error, for example.)

Reply:
* `BadTargetLabel` is returned by create syscalls when the new direntry's label (i.e. target label) cannot flow to the function's label*. The previous implmentation just used `Unauthorized` instead of a separate error. I think separating the error out is more informative.
* For `UidCollision`, I agree. I'll make the change.

*Recall that the create syscalls' semantics are reading parent directories one by one starting with the root directory during which the function's label gets raised if needed, then checking if the function is authorized to write after reading the direct parent directory, and last checking if the function's label at the moment can flow to the target label. The intention of final check is to prevent functions from creating objects they cannot write in the future.

Reply:
made [the change](https://github.com/princeton-sns/snapfaas/pull/31/commits/e40e4d6498a53b0b465fe2a114daed253787e37c) to handle uid collisions inside the file system.